Lib/test/support/_hypothesis_stubs/_helpers.py

```diff
@@ -0,0 +1,34 @@
+# Stub out only the subset of the interface that we actually use in our tests.
+class StubClass:
+    def __init__(self, *args, **kwargs):
+        super().__init__()
+        self.__stub_args = args
+        self.__stub_kwargs = kwargs
+
+    def __repr__(self):
+        argstr = ", ".join(self.__stub_args)
+        kwargstr = ", ".join(
+            f"{kw}={val}" for kw, val in self.__stub_kwargs.items()
+        )
+
+        in_parens = argstr
+        if kwargstr:
+            in_parens += ", " + kwargstr
+
+        return f"{self.__qualname__}({in_parens})"
```

Comment:
Since the various `.map()`, `.filter()`, `.__or__()` methods don't alter the repr at all, I'd probably prefer that this be less specific - better to represent `integers().map(str)` as `<stub object at 0x...>` than `integers()`!

I also don't think it's worth the code to make the repr track all the methods, unless adding a `Stub.__trailing_repr` argument would work?

Reply:
OK, this should be working now. Here's a little test file:

```python
from test.support.hypothesis_helper import hypothesis

def complement(x):
    return not x

def identity(x):
    return x

st = hypothesis.strategies

test_strategies = [
    st.integers(),
    st.floats().filter(complement),
    st.lists(st.booleans).flatmap(identity),
    st.characters() | st.dates().map(lambda x: x),
    st.booleans() | st.uuids(),
]

for test_strategy in test_strategies:
    print(repr(test_strategy))
```

Results:

```
hypothesis.strategies.integers()
hypothesis.strategies.floats().filter(complement)
hypothesis.strategies.lists().flatmap(identity)
one_of(hypothesis.strategies.characters(), hypothesis.strategies.dates().map(<lambda>))
one_of(hypothesis.strategies.booleans(), hypothesis.strategies.uuids())
```